vllm/model_executor/layers/sampler.py

```diff
@@ -35,8 +36,8 @@ def __init__(self,
                  org_vocab_size: Optional[int] = None) -> None:
         super().__init__()
         self.vocab_size = vocab_size
-        # Transformers-neuronx generate outputs as logits directly.
-        self.logits_as_hidden_states = is_neuron()
+        # Transformers-neuronx / OpenVINO optimum-intel generate outputs as logits directly.
+        self.logits_as_hidden_states = is_neuron() or is_openvino_optimum_intel()
```

Comment:
@slyalin do you think it's possible to remove `is_neuron()` and `is_openvino_optimum_intel()` here and detect scenario based on shapes of `hidden_states` ?